perceval/backends/mozilla/mozillaclub.py

```diff
@@ -0,0 +1,397 @@
+
+        :returns: this backend supports items resuming
+        """
+        return False
+
+    @staticmethod
+    def metadata_id(item):
+        """Extracts the identifier from an event item."""
+
+        try:
+            a = item.keys()
+        except AttributeError:
+            print ("->"+item+"<-")
+            raise
+        return str(item['Date of Event']+"_"+item['Club Name'])
+
+    @staticmethod
+    def metadata_category(item):
+        """Extracts the category from a item.
+
+        This backend only generates one type of item which is
+        'event'.
+        """
+        return 'event'
+
+    @staticmethod
+    def metadata_updated_on(item):
+        """Extracts the update time from a MozillaClub item.
+
+        The timestamp is extracted from 'end' field.
+        This date is a UNIX timestamp but needs to be converted to
+        a float value.
+
+        :param item: item generated by the backend
+
+        :returns: a UNIX timestamp
+        """
+        date = str_to_datetime(item['Date of Event'])
+        return float(date.timestamp())
+
+
+class MozillaClubClient:
+    """MozillaClub API client.
+
+    This class implements a simple client to retrieve events from
+    projects in a MozillaClub site.
+
+    :param url: URL of MozillaClub
+
+    :raises HTTPError: when an error occurs doing the request
+    """
+
+    def __init__(self, url):
+        self.url = url
+
+    def call(self, uri):
+        """Run an API command.
+
+        :param params: dict with the HTTP parameters needed to run
+            the given command
+        """
+        logger.debug("MozillaClub client calls API: %s", self.url)
+
+        req = requests.get(uri)
+        req.raise_for_status()
+
+        return req.text
+
+    def get_cells(self):
+        """Retrieve all cells from the spreadsheet."""
+
+        logging.info("Retrieving all cells spreadsheet data ...")
+        raw_cells = self.call(self.url)
+
+        return raw_cells
+
+
+class MozillaClubParser:
+    """Git log parser.
+
+    This class parses a string in JSON format from a Google Spreadsheet. The
+    feed includes all the cells in a list.
+
+    Events are rows in the spreadsheet. The columns are the fields
+    for the event. The JSON retrieved from the spreadsheet feed is
+    a plain list with all the cells from all the rows.
+    """
+    def parse(self, sheet_json):
+        """Parse the MozillaClub spreadsheet feed cells json."""
+
+        nevents_wrong = 0
+
+        cells = sheet_json['feed']['entry']
+
+        event_fields = self.__get_event_fields(cells)
+
+        # Process all events reading the rows according to the event template
+        # The only way to detect the end of row is looking to the
+        # number of column. When the max number is reached (cell_cols) the next
+        # cell is from the next row.
+        # The first while is for looping rows and the second columns
+        while cells:
```

Comment:
I don't like the whole idea of this loop. You're using `__get_next_event` to change the contents of `cells`. This is not a good practice because it is hard to follow what the code is doing. Methods should not change the status of an object that is given as parameter.

One possible solution is to create an Interable object which is initialized with the cells and which iteraters over cells, parsing them.